File: JS/backend/models/estacionSismologicaModel.js
import { DataTypes, Model } from "sequelize";
import sequelize from "../data/db.js";
import SismografoModel from "./sismografoModel.js";

class EstacionSismologicaModel extends Model {}

EstacionSismologicaModel.init({
    codigo: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
    nombre: { type: DataTypes.STRING, allowNull: false },
    latitud: { type: DataTypes.FLOAT, allowNull: false },
    longitud: { type: DataTypes.FLOAT, allowNull: false },
    fechaSolicitudCertificacion: { type: DataTypes.DATE },
    docCertificacion: { type: DataTypes.STRING },
    nroCertificacionAdquisicion: { type: DataTypes.STRING },
}, {
    sequelize,
    modelName: "EstacionSismologica",
    tableName: "EstacionesSismologicas",
    timestamps: false,
    underscored: true
})

EstacionSismologicaModel.hasOne(SismografoModel, {
    foreignKey: "codigoEstacion",   // Foreign key in SismografoModel
    as: "sismografo"    // Alias for the association
})



export default EstacionSismologicaModel;

File: persistencia/estacionSismologicaModel.py
from peewee import *
from RUTAS.rutas import ARCHIVO_BD

# Conexión a la base de datos SQLite
db = SqliteDatabase(ARCHIVO_BD)


class EstacionSismologicaModel(Model):
    codigo_estacion = CharField(primary_key=True)
    nombre = CharField()
    latitud = FloatField()
    longitud = FloatField()
    fecha_solicitud_certificacion = DateTimeField(null=True)
    documento_certificacion = CharField(null=True)
    numero_certificacion = CharField(null=True)

    class Meta:
        database = db
        table_name = "EstacionSismologica"
